src/SharedInfrastructure/Guard.cs

```diff
@@ -20,16 +21,9 @@ internal static partial class Guard
     /// <typeparam name="TValue">The type of the value.</typeparam>
     /// <exception cref="ArgumentNullException"><paramref name="value"/> is null.</exception>
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
-    public static void NotNull<TValue>(TValue value, string parameterName)
-        where TValue : class
-    {
-        if (value is not null)
-        {
-            return;
-        }
-
-        ThrowHelper.ThrowArgumentNullExceptionForNotNull(parameterName);
-    }
+    public static void NotNull<TValue>([NotNull]TValue? value, string parameterName)
```

Comment:
We should delete `ThrowHelper.ThrowArgumentNullExceptionForNotNull(...)` now that it is not in use.